Sources/Core/Error/UIError.swift

```diff
@@ -5,36 +5,41 @@
 
 import SwiftUI
 
+/// A model for user-facing error messages to be displayed in the UI
+///
+/// Use `DefaultUIError` or implement your own to have more rich
+/// data types including images, titles, etc.
+public protocol UIError: Error, Identifiable {
+    var message: String { get }
+    var isRetryable: Bool { get }
+}
 
+public extension UIError {
+    
+    /// Override this if needed, for example, if
+    /// your error implementation includes a title.
+    var id: String { message + "_\(isRetryable)" }
+}
 
-/// A model for user-facing error messages to be displayed in the UI
-public struct UIError: Error, Hashable {
+/// This is a default, minimal implementation. If you need titles, images, etc.
+/// you can provide your own implementation of `UIError`.
+public struct DefaultUIError: UIError {
 
-    public static func `default`(isRetryable: Bool) -> UIError {
-        UIError(
+    public static func `default`(isRetryable: Bool) -> some UIError {
```

Comment:
I'm not sure why this was originally returning `UIError`, but why not return `DefaultUIError` since the function is literally defined on `DefaultUIError`?